File: solutions/ruby/scrabble-score/1/scrabble_score.rb
class Scrabble
  @@scores = {}

  def self.add(sc, letters)
    letters.each_char {@@scores[_1] = sc}
  end

  add(1, "aeioulnrst")
  add(2, "dg")
  add(3, "bcmp")
  add(4, "fhvwy")
  add(5, "k")
  add(8, "jx")
  add(10, "qz")

  def initialize (str)
    @str =  str
    debug @@scores[3]
  end
  
  def score()
    res = 0
    @str.each_char do |ch|
      res += @@scores[ch.downcase]
    end
    res
  end
end

File: solutions/julia/scrabble-score/1/scrabble-score.jl
scores = Dict()
for (nu,letters) in [(1,"AEIOULNRST"),(2,"DG"),(3,"BCMP"),(4,"FHVWY"),(5,"K"),(8,"JX"),(10,"QZ")]
    for ltr in letters
        scores[ltr]=nu
    end
end
function score(str)
    total = 0
    for ltr in str
        if ltr>='a' && ltr<='z'
            ltr-=32
        end
        if ltr>='A' && ltr<='Z'
            total+=scores[ltr]
        end
    end
    return total
end
